### DISCUSSION TITLE
Won't haves

### DISCUSSION THREAD

**Discussion by jhpratt** [rust-marker/design#4] [Ideas]
Use this discussion for suggestions of "won't haves" for a custom linter. Please limit suggestions to one per top-level comment to allow for easier upvoting/downvoting and replies.

**Comment by jhpratt** [comment]
Lints will not have access to the compiler's or linter's internals.

**Reply by jhpratt** [reply]
Accepting this without objection as it seems unlikely we _would_ want to do this.

**Comment by jhpratt** [comment]
Lints will not have access to any system APIs, including file or network access.

**Reply by xFrednet** [reply]
Do we want to give the lints access to environment values?

**Reply by jplatte** [reply]
What about configurable lints? Clippy reads a bunch of settings from `clippy.toml`, for example the MSRV (which it might soon read from `Cargo.toml` too).

**Reply by xFrednet** [reply]
Related #7

**Reply by jhpratt** [reply]
I'd like to clarify this as _direct_ access. The possibility of the linter providing an API to inspect the manifest seems reasonable and I don't want to rule it out. Likewise for environment variables.

**Reply by jhpratt** [reply]
Let's agree on this one? The linter shouldn't provide access to the network, file system, or similar, but if the lint implementation isn't sandboxed then they could of course import the methods themselves. I don't think that capability should be guaranteed, though.

**Reply by xFrednet** [reply]
Yes, I think we can all agree with this :+1:

**Reply by DevinR528** [reply]
Sounds good to me!

**Reply by jhpratt** [reply]
Added.
